TombEngine/Game/Lara/lara_one_gun.cpp

```diff
@@ -495,20 +505,26 @@ void DrawShotgun(ItemInfo& laraItem, LaraWeaponType weaponType)
 		ReadyShotgun(laraItem, weaponType);
 	}
 
-	player.LeftArm.FrameBase = player.RightArm.FrameBase = GetAnimData(weaponItemPtr).FramePtr;
-	player.LeftArm.FrameNumber = player.RightArm.FrameNumber = weaponItemPtr->Animation.FrameNumber - GetAnimData(weaponItemPtr).frameBase;
-	player.LeftArm.AnimNumber = player.RightArm.AnimNumber = weaponItemPtr->Animation.AnimNumber;
+	player.LeftArm.AnimObjectID =
+	player.RightArm.AnimObjectID = GetWeaponObjectID(weaponType);
+	player.LeftArm.AnimNumber =
+	player.RightArm.AnimNumber = weaponItemPtr->Animation.AnimNumber;
+	player.LeftArm.FrameNumber =
+	player.RightArm.FrameNumber = weaponItemPtr->Animation.FrameNumber;
 }
 
 void UndrawShotgun(ItemInfo& laraItem, LaraWeaponType weaponType)
 {
-	auto& player = *GetLaraInfo(&laraItem);
-
+	auto& player = GetLaraInfo(laraItem);
 	auto& item = g_Level.Items[player.Control.Weapon.WeaponItem];
+
+	// HACK
+	if (item.Animation.TargetState != WEAPON_STATE_UNDRAW && item.Animation.ActiveState != WEAPON_STATE_UNDRAW)
+		SetAnimation(item, (weaponType == LaraWeaponType::GrenadeLauncher) ? 5 : 3);
+
```

Comment:
Why?

Reply:
The legacy weapons subsystem is a hideous mess, so I had to hack my way through. Don't mind any of it. The whole thing needs a rewrite from scratch. :)